akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/sse/LineParser.scala

```diff
@@ -58,7 +58,10 @@ private final class LineParser(maxLineSize: Int) extends GraphStage[FlowShape[By
               case _ ⇒
                 parseLines(bs, from, at + 1, parsedLines)
             }
-        buffer = parseLines(buffer ++ grab(in)) match {
+
+        // start the search where it ended, prevent iterating over all the buffer again
+        val currentBufferStart = if (buffer.isEmpty) 0 else buffer.length - 1
```

Comment:
Do we have tests for all combinations of having the chunk boundary directly before and after `CR` and `LF` and when it happens to be between `CR` and `LF`?

Reply:
Could you elaborate a little more about it?

Reply:
All of this logic is about matching CR LF pairs, so this change has the risk that we might now miss a pair or match it wrongly (e.g. adding an extra CR to the first element). We might already have those tests, if not we need to add them.